packages/pg-vector-selector/src/utils.ts

```diff
@@ -1,4 +1,8 @@
-import type { NonNegativeInteger } from "type-fest";
+import type { GreaterThan, Integer } from "type-fest";
+
+/** type function to check if a number is greater than 0 */
+type GreaterThanZeroInteger<T extends number> = GreaterThan<Integer<T>, 0> extends true ? T : never;
```

Comment:
This is my hobby! But it is fun tho.
I hope this helps! haha!

Reply:
Actually I need to write type-check test for this type function. However this project is commonjs project so we cannot use type-test with vitest I'm afraid!

Reply:
if the function only accepts NonNegativeInteger including 0, we can keep it.
But we don't accept 0, so I defined this(maybe I'm too strict?lol)